src/core/include/openvino/core/type.hpp

```diff
@@ -101,6 +101,12 @@ is_type(Value value) {
     return value && value->get_type_info().is_castable(Type::get_type_info_static());
 }
 
+/// \brief Tests if value is a pointer/shared_ptr that can be statically cast to any of the specified types
+template <typename Type, typename... Types, typename Value>
+bool is_type_any_of(Value value) {
```

Comment:
Move it to developer API it can internal tool

Reply:
Moved to `dev_api`